src/RequestExecutor/HttpRequestExecutor.ts

```diff
@@ -474,4 +468,84 @@ export class HttpRequestExecutor implements RequestExecutor {
 
     return new Request(result);
   }
+
+  private certificatesForRequest(request: Request): Cert[] {
+    const cachedCertificate = this.certificatesCache.get(request);
+    if (cachedCertificate) {
+      return [cachedCertificate];
+    }
+
+    const requestUrl = new URL(request.url);
+    const port = Helpers.portFromURL(requestUrl);
+
+    return this.options.certs.filter((cert: Cert) =>
+      Helpers.matchHostnameAndPort(requestUrl.hostname, port, cert)
+    );
+  }
+
+  private async tryRequest(request: Request): Promise<Response | undefined> {
+    const { res, body } = await this.request(request);
+
+    logger.trace(
+      'received following response for request %j: headers: %j body: %s',
+      {
+        url: request.url,
+        protocol: this.protocol,
+        method: request.method
+      },
+      {
+        statusCode: res.statusCode,
+        headers: res.headers
+      },
+      body.slice(0, 500).concat(body.length > 500 ? '...' : '')
+    );
+
+    return new Response({
+      body,
+      protocol: this.protocol,
+      statusCode: res.statusCode,
+      headers: res.headers,
+      encoding: request.encoding
+    });
+  }
+
+  private async tryRequestWithCertificate(
+    request: Request,
+    cert: Cert
+  ): Promise<Response | undefined> {
+    await request.setCert(cert);
+
+    // eslint-disable-next-line @typescript-eslint/return-await
+    return await this.tryRequest(request);
+  }
+
+  private async tryRequestWithCertificates(
+    request: Request,
+    certs: Cert[]
+  ): Promise<Response> {
+    const requestsWithCerts: Promise<Response>[] = certs.map(
+      async (cert: Cert) => {
+        logger.debug(
+          'Executing HTTP request with following params: %j',
+          request
+        );
+        try {
+          const response = await this.tryRequestWithCertificate(request, cert);
+          this.certificatesCache.add(request, cert);
+
+          return response;
+        } catch (error) {
+          const msg = Helpers.isTlsCertError(error)
+            ? `Failed to do successful request with certificate ${cert.path}. It will be excluded from list of known certificates.`
+            : `Unexpected error occured during request: ${error}`;
+          logger.warn(msg);
+          throw Error(msg);
+        }
+      }
+    );
+
+    // Important to wait for valid response.
+    // eslint-disable-next-line @typescript-eslint/return-await
+    return await Promise.any(requestsWithCerts);
```

Comment:
Promises are chained, this await just does not make any sense. This is explicitly flagged by linter 

```suggestion
    return Promise.any(requestsWithCerts);
```